File: tldjfj123/Week07/성격유형검사하기/Solution.java
package 성격유형검사하기;

import java.util.Arrays;
import java.util.HashMap;

class Solution {
    public String solution(String[] survey, int[] choices) {
        //RCJA
        int[] table1 = new int[4];
        //TFMN
        int[] table2 = new int[4];

        //"RT", "CF", "JM", "AN"

        for (int i = 0; i < survey.length; i++) {
            String keyword = survey[i];
            int score = choices[i];

            if (keyword.equals("RT")) {
                if (score < 4) {
                    table1[0] += (4 - score);
                } else {
                    table2[0] += (score - 4);
                }
            } else if (keyword.equals("TR")) {
                if (score > 4) {
                    table1[0] += (score - 4);
                } else {
                    table2[0] += (4 - score);
                }
            } else if (keyword.equals("CF")) {
                if (score < 4) {
                    table1[1] += (4 - score);
                } else {
                    table2[1] += (score - 4);
                }
            } else if (keyword.equals("FC")) {
                if (score > 4) {
                    table1[1] += (score - 4);
                } else {
                    table2[1] += (4 - score);
                }
            } else if (keyword.equals("JM")) {
                if (score < 4) {
                    table1[2] += (4 - score);
                } else {
                    table2[2] += (score - 4);
                }
            } else if (keyword.equals("MJ")) {
                if (score > 4) {
                    table1[2] += (score - 4);
                } else {
                    table2[2] += (4 - score);
                }
            } else if (keyword.equals("AN")) {
                if (score < 4) {
                    table1[3] += (4 - score);
                } else {
                    table2[3] += (score - 4);
                }
            } else {
                if (score > 4) {
                    table1[3] += (score - 4);
                } else {
                    table2[3] += (4 - score);
                }
            }
        }

        String res = "";

        if (table1[0] >= table2[0]) {
            res += "R";
        } else {
            res += "T";
        }

        if (table1[1] >= table2[1]) {
            res += "C";
        } else {
            res += "F";
        }

        if (table1[2] >= table2[2]) {
            res += "J";
        } else {
            res += "M";
        }

        if (table1[3] >= table2[3]) {
            res += "A";
        } else {
            res += "N";
        }


        return res;

    }
}

File: tldjfj123/Week07/성격유형검사하기/Solution.py
def solution(survey, choices):
    # R / T, C / F, J / M, A / N
    table1 = {"R" : 0, "C" : 0, "J" : 0, "A" : 0}
    table2 = {"T" : 0, "F" : 0, "M" : 0, "N" : 0}
    
    for i in range(len(survey)) :
        keyword = survey[i]
        score = choices[i]
        if list(keyword) == sorted(keyword) : # 그대로
            if score < 4 :
                table1[keyword[0]] += (4 - score)
            else :
                table2[keyword[1]] += (score - 4)
        else : # 반대
            if score > 4 :
                table1[keyword[1]] += (score - 4)
            else :
                table2[keyword[0]] += (4 - score)
        
    key1 = list(table1.keys())
    key2 = list(table2.keys())
    
    res = []
    
    for i in range(len(key1)) :
        if table1[key1[i]] >= table2[key2[i]] :
            res.append(key1[i])
        else :
            res.append(key2[i])
    
    return ''.join(res)
            
            
    
        
        